### PR TITLE
avoid re-render of MapLibreLayer when updating any layer of the map

### LINKED ISSUE #2588: maplibre layer is rendered each time a layer on the map changes

### PR DESCRIPTION
**Pull request by taulinger**
Important: Due to these changes the collection retrieved from `map.getLayers()` does not reflect the order of the layers on the ol.Map anymore!

@thiloSchlemmer 
Would you please have a look if this touches the following lines
https://github.com/ldbv-by/bav4/blob/94dcade71ea4446396085e24739409976edd5517/src/modules/olMap/handler/measure/OlMeasurementHandler.js#L123

https://github.com/ldbv-by/bav4/blob/94dcade71ea4446396085e24739409976edd5517/src/modules/olMap/handler/draw/OlDrawHandler.js#L156

fixes #2588

### CODE REVIEW

**Review by thiloSchlemmer — APPROVED** [review]
THX!

**Review by thiloSchlemmer — COMMENTED** [review]
i updated the layer handling in the two mentioned OlLayerHandlers...
the rest looks good to me!

File: src/modules/olMap/handler/measure/OlMeasurementHandler.js [hunk 1]
```diff
@@ -694,7 +701,7 @@ export class OlMeasurementHandler extends OlLayerHandler {
 			// register the stored data as new georesource
 			this._geoResourceService.addOrReplace(vgr);
 			const layerId = this._layerId ?? `${id}_draw`;
-			addLayer(layerId, { geoResourceId: id, constraints: { metaData: false } });
+			addLayer(layerId, { zIndex: this._layerZIndex ?? 0, geoResourceId: id, constraints: { metaData: false } });
```
Comment (taulinger) [inline src/modules/olMap/handler/measure/OlMeasurementHandler.js:704]: see https://github.com/ldbv-by/bav4/pull/2586/files#r1782883763

**Review by thiloSchlemmer — APPROVED** [review]
I've fixed the mentioned lines.

**Pull request merged**